File: js/0557.0_Reverse_Words_in_a_String_III.js
/*
执行用时：56 ms, 在所有 JavaScript 提交中击败了99.61% 的用户
内存消耗：47 MB, 在所有 JavaScript 提交中击败了82.19% 的用户
通过测试用例：29 / 29
*/

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let arr = s.split(' ');
    for (let i = 0; i < arr.length; i++) {
        let num = arr[i].split('').reverse().join('');
        arr[i] = num;
    }
    return arr.join(' ');
};


/*
执行用时：72 ms, 在所有 JavaScript 提交中击败了73.81% 的用户
内存消耗：47.1 MB, 在所有 JavaScript 提交中击败了70.57% 的用户
通过测试用例：29 / 29
*/
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    return s.split(' ').map(item => item.split('').reverse().join('')).join(' ')
};

File: problems/0557.0_Reverse_Words_in_a_String_III.py
'''
Runtime: 40 ms, faster than 90.84% of Python3 online submissions for Reverse Words in a String III.
Memory Usage: 14.4 MB, less than 97.19% of Python3 online submissions for Reverse Words in a String III.
'''
class Solution:
    def reverseWords(self, s: str) -> str:
        return ' '.join(word[:: -1] for word in s.split())
        